### PR TITLE
 Class to handle results of Promotion Quiz and unit tests

### PR DESCRIPTION
**Pull request by lzoebaker**
Class to handle results of Promotion Quiz and unit tests

### CODE REVIEW

File: src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.google.sps.data;
+
+import java.util.List;
+
+/** Returns the recommended career path based off the career quiz results. */
+public class ProcessPromotionQuizResults {
+  private ProcessPromotionQuizResults() {}
+
+  public static Boolean getIsPromotedOrNot(List<QuestionChoice> userChoices, double threshold)
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java:9]: name this method "isPromoted"

File: src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      throws IllegalArgumentException {
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java:10]: Include a message in the exception so we know what went wrong

File: src/test/java/com/google/sps/PromotionQuizResultsTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            new QuestionChoice(CHOICE_3, CAREER_2, /* isAcceptableChoice = */ false),
+            new QuestionChoice(CHOICE_4, CAREER_3, /* isAcceptableChoice = */ false));
+    double threshold = 0.5;
+    Boolean expected = true;
+    Boolean result = ProcessPromotionQuizResults.getIsPromotedOrNot(userChoices, threshold);
+    Assert.assertEquals(result, expected);
+  }
+
+  @Test
+  public void proportionAcceptableChoicesAreOverThreshold_DoesNotGetPromoted() {
```
Comment (WillHipschman) [inline src/test/java/com/google/sps/PromotionQuizResultsTest.java:57]: Should this be "...ChoicesAreUnderThreshold"?

File: src/test/java/com/google/sps/PromotionQuizResultsTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            new QuestionChoice(CHOICE_2, CAREER_2, /* isAcceptableChoice = */ false),
+            new QuestionChoice(CHOICE_3, CAREER_2, /* isAcceptableChoice = */ false),
+            new QuestionChoice(CHOICE_4, CAREER_3, /* isAcceptableChoice = */ true));
+    double threshold = 0.4;
+    Boolean expected = false;
+    Boolean result = ProcessPromotionQuizResults.getIsPromotedOrNot(userChoices, threshold);
+    Assert.assertEquals(result, expected);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```
Comment (WillHipschman) [inline src/test/java/com/google/sps/PromotionQuizResultsTest.java:70]: Is there a way to assert that the exception message is what we expect?

File: src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java (hunk 1 above)
Comment (DirtGrubDylan) [reply src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java:9]: +1

File: src/test/java/com/google/sps/PromotionQuizResultsTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  private static final String CHOICE_2 = "choice2";
+  private static final String CHOICE_3 = "choice3";
+  private static final String CHOICE_4 = "choice4";
+  private static final String CHOICE_5 = "choice5";
+
+  @Test
+  public void proportionAcceptableChoicesAreOverThreshold_GetsPromoted() {
+    List<QuestionChoice> userChoices =
+        Arrays.asList(
+            new QuestionChoice(CHOICE_1, CAREER_1, /* isAcceptableChoice = */ true),
```
Comment (DirtGrubDylan) [inline src/test/java/com/google/sps/PromotionQuizResultsTest.java:32]: I little tip, if you name two static variables:
```
private static final ACCEPTABLE_CHOICE = true;
private static final NOT_ACCEPTABLE_CHOICE = false;
```

Then you can do 
```
new QuestionChoice(CHOICE_1, CAREER_1, ACCEPTABLE_CHOICE)
```

File: src/test/java/com/google/sps/PromotionQuizResultsTest.java (hunk 3 above)
Comment (DirtGrubDylan) [reply src/test/java/com/google/sps/PromotionQuizResultsTest.java:57]: +1

File: src/test/java/com/google/sps/PromotionQuizResultsTest.java (hunk 4 above)
Comment (lzoebaker) [reply src/test/java/com/google/sps/PromotionQuizResultsTest.java]: Yes, found a way! Changed.

File: src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java (hunk 2 above)
Comment (lzoebaker) [reply src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java:13]: Added.

File: src/test/java/com/google/sps/PromotionQuizResultsTest.java (hunk 3 above)
Comment (lzoebaker) [reply src/test/java/com/google/sps/PromotionQuizResultsTest.java]: Oops, fixed.

File: src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java (hunk 1 above)
Comment (lzoebaker) [reply src/main/java/com/google/sps/data/ProcessPromotionQuizResults.java]: Changed.

File: src/test/java/com/google/sps/PromotionQuizResultsTest.java (hunk 5 above)
Comment (lzoebaker) [reply src/test/java/com/google/sps/PromotionQuizResultsTest.java]: Nifty, changed.

**Pull request merged**